### DISCUSSION TITLE
Converting UltraHDR to HEIC with gain map

### DISCUSSION THREAD

**Discussion by Tumist76** [ImageMagick/ImageMagick#7598] [Help]
### ImageMagick version

7.1.1-38

### Operating system, version and so on

MacOS 14.5 (23F79)

### Description

Is it possible to convert UltraHDR jpgs to HEIC without losing HDR gain map?

I've tried using 
```
magick mogrify -format heic uhdr:/Users/tumist/Downloads/PXL_20240830_180807425.NIGHT.jpg
```

but it outputted a HEIC without HDR gain map, at least without iOS/MacOS supported HDR.

Attaching UltraHDR for reference.
![PXL_20240830_180807425 NIGHT](https://github.com/user-attachments/assets/d3048596-2b90-4826-aefb-7be3570c1c53)

**Comment by fmw42** [comment]
Magick mogrify is old.  Please try with just magick. 

One of the IM developers will need to comment if the gain map is transferred.

**Comment by urban-warrior** [comment]
Try this command:

```
magick uhdr:PXL_20240830_180807425.NIGHT.jpg PXL_20240830_180807425.NIGHT.heic
```

Also, try `mpo:PXL_20240830_180807425.NIGHT.jpg`.

**Reply by Tumist76** [reply]
Tried both commands, sadly, they produces non-HDR HEICs. 
Seeing this issue, it seems like IM doesn't support gain maps for now :(
https://github.com/ImageMagick/ImageMagick/issues/6377

**Reply by fmw42** [reply]
Gain maps only in JPG, as far as I know. You asked to carry the gain map over, not to apply it. Did I misunderstand?  A developer will need to comment further.

**Comment by isenberg** [comment]
I also tried to produce other HDR images than UltraHDR JPEG, for example AVIF, but failed to add the gainmap. But creating UHDR JPEG with gain-map works fine now since at least last week https://github.com/ImageMagick/ImageMagick/pull/7635 .
